diff -Nurd wxWidgets-2.9.2/src/osx/utils_osx.cpp wxWidgets-2.9.2.patched/src/osx/utils_osx.cpp
--- wxWidgets-2.9.2/src/osx/utils_osx.cpp	2011-07-04 17:26:11.000000000 -0400
+++ wxWidgets-2.9.2.patched/src/osx/utils_osx.cpp	2011-08-29 12:44:35.000000000 -0400
@@ -50,6 +50,10 @@
 #endif
 #endif
 
+/* for 10.7 */
+size_t displayBitsPerPixel (CGDirectDisplayID displayId);
+
+
 // Check whether this window wants to process messages, e.g. Stop button
 // in long calculations.
 bool wxCheckForInterrupt(wxWindow *WXUNUSED(wnd))
@@ -69,7 +73,7 @@
 // Returns depth of screen
 int wxDisplayDepth()
 {
-    int theDepth = (int) CGDisplayBitsPerPixel(CGMainDisplayID());
+    int theDepth = (int) displayBitsPerPixel(CGMainDisplayID());
     return theDepth;
 }
 
@@ -244,4 +248,26 @@
     return wxString( (char*) &from[1] , wxConvLocal , from[0] );
 }
 
+//---------------------------------------------------------------------------
+// Lion Specific display depth finder
+//
+
+size_t displayBitsPerPixel (CGDirectDisplayID displayId) {
+
+	CGDisplayModeRef mode = CGDisplayCopyDisplayMode(displayId);
+	size_t depth = 0;
+
+	CFStringRef pixEnc = CGDisplayModeCopyPixelEncoding(mode);
+	if(CFStringCompare(pixEnc, CFSTR(IO32BitDirectPixels), kCFCompareCaseInsensitive) == kCFCompareEqualTo)
+		depth = 32;
+	else if(CFStringCompare(pixEnc, CFSTR(IO16BitDirectPixels), kCFCompareCaseInsensitive) == kCFCompareEqualTo)
+		depth = 16;
+	else if(CFStringCompare(pixEnc, CFSTR(IO8BitIndexedPixels), kCFCompareCaseInsensitive) == kCFCompareEqualTo)
+		depth = 8;
+
+		CFRelease(pixEnc);
+		CFRelease(mode);
+		return depth;
+}
+
 #endif // wxOSX_USE_COCOA_OR_CARBON
